lib/tiniergltf/tiniergltf.hpp

```diff
@@ -1252,10 +1258,12 @@ struct GlTF {
 			const BufferView &view = bufferViews->at(bufferView);
 			if (view.byteStride.has_value())
 				check(*view.byteStride % accessor.componentSize() == 0);
-			check(byteOffset < view.byteLength);
-			// Use division to avoid overflows.
+
 			const auto effective_byte_stride = view.byteStride.value_or(accessor.elementSize());
-			check(count <= (view.byteLength - byteOffset) / effective_byte_stride);
+			// Want: (count-1) * effective_byte_stride + accessor.elementSize() + byteOffset <= view.byteLength
+			// so the last item is still in-bounds and can be read. Be careful to avoid wraparound.
+			const auto remaining_bytes = checkedSub(checkedSub(view.byteLength, byteOffset), accessor.elementSize());
+			check((remaining_bytes/effective_byte_stride) + 1 >= count);
```

Comment:
Where does the +1 (or -1 in the comment above) come from? I don't know much about this code, and this generic function makes it quite hard to comprehend what we're doing here. Thus, I'd appreciate it if you could add a few in-line comments to help better understanding.

Reply:
alright I've tried to refactor this a little: https://github.com/luanti-org/luanti/pull/16590/commits/ad0c19bdb3347d9b2f3fa8eecafb01003adc31de